From a291542e32031b1416ca27a6db49dfe3ed44e340 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Tue, 23 Apr 2019 15:36:19 +0300
Subject: [PATCH 2/5] plat: renesas: Fix 'rcar_bl31_save_timer_state' defined
 but not used
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This puts rcar_bl31_save_timer_state function under
the RCAR_SYSTEM_SUSPEND ifdef to fix the following
compilation error when RCAR_SYSTEM_SUSPEND is disabled:

  error: ‘rcar_bl31_save_timer_state’ defined but not used

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 plat/renesas/rcar/drivers/pwrc/rcar_pwrc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/plat/renesas/rcar/drivers/pwrc/rcar_pwrc.c b/plat/renesas/rcar/drivers/pwrc/rcar_pwrc.c
index fb27a02..b0f0fc4 100644
--- a/plat/renesas/rcar/drivers/pwrc/rcar_pwrc.c
+++ b/plat/renesas/rcar/drivers/pwrc/rcar_pwrc.c
@@ -167,9 +167,9 @@ RCAR_INSTANTIATE_LOCK
 #if RCAR_SYSTEM_SUSPEND
 static void rcar_bl31_set_self_refresh(void);
 static void rcar_bl31_set_self_refresh_e3(void);
+static void rcar_bl31_save_timer_state(void);
 #endif /* RCAR_SYSTEM_SUSPEND */
 static void SCU_power_up(uint64_t mpidr);
-static void rcar_bl31_save_timer_state(void);
 
 static uint64_t rcar_bl31_saved_cntpct_el0;
 static uint32_t rcar_bl31_saved_cntfid;
@@ -933,6 +933,7 @@ uint32_t rcar_bl31_get_cpu_num(uint32_t cluster_type)
 	return num;
 }
 
+#if RCAR_SYSTEM_SUSPEND
 static void rcar_bl31_save_timer_state(void)
 {
 	rcar_bl31_saved_cntpct_el0 = read_cntpct_el0();
@@ -940,6 +941,7 @@ static void rcar_bl31_save_timer_state(void)
 	rcar_bl31_saved_cntfid =
 		mmio_read_32((uintptr_t)(RCAR_CNTC_BASE + RCAR_CNTFID_OFF));
 }
+#endif
 
 void rcar_bl31_restore_timer_state(void)
 {
-- 
2.7.4

